control-plane/gateways/deployment_dataplane_container.go

```diff
@@ -112,11 +109,10 @@ func consulDataplaneContainer(config GatewayConfig, containerConfig *v2beta1.Gat
 	wanPort := corev1.ContainerPort{
 		Name:          "wan",
 		ContainerPort: int32(constants.DefaultWANPort),
+		HostPort:      containerConfig.HostPort,
 	}
 
-	if containerConfig != nil {
-		wanPort.ContainerPort = int32(443 + containerConfig.PortModifier)
-	}
+	wanPort.ContainerPort = 443 + containerConfig.PortModifier
```

Comment:
Just ranting, why is hostPort int32 and containerPort just int?

Reply:
It is int32, I updated the value to match what the object is expecting. https://github.com/hashicorp/consul-k8s/pull/3381/files#diff-3a9113026018fc0f58d311cb39bf0de995570740f1ebde6fa3cb1dcbb4e2c64bR101

Reply:
nice!